sources/packages/backend/apps/api/src/services/student-assessment/student-assessment.service.ts

```diff
@@ -236,4 +246,118 @@ export class StudentAssessmentService extends RecordDataModelService<StudentAsse
       .addOrderBy("assessment.submittedDate", "DESC")
       .getMany();
   }
+
+  /**
+   * Creates a manual assessment for the application.
+   * @param applicationId application id.
+   * @param note note.
+   * @param userId user id who triggered the manual reassessment.
+   */
+  async manualReassessment(
+    applicationId: number,
+    note: string,
+    userId: number,
+  ): Promise<void> {
+    return this.dataSource.transaction(async (transactionalEntityManager) => {
+      const currentUser = { id: userId } as User;
+      const now = new Date();
+      const applicationRepo =
+        transactionalEntityManager.getRepository(Application);
+      const application = await applicationRepo.findOne({
+        select: {
+          id: true,
+          currentAssessment: {
+            id: true,
+            offering: {
+              id: true,
+            },
+            studentAppeal: {
+              id: true,
+            },
+          },
+          studentAssessments: {
+            id: true,
+            studentAssessmentStatus: true,
+          },
+          student: { id: true },
+        },
+        relations: {
+          currentAssessment: { offering: true, studentAppeal: true },
+          studentAssessments: true,
+          student: true,
+        },
+        where: {
+          id: applicationId,
+          studentAssessments: {
+            triggerType: AssessmentTriggerType.OriginalAssessment,
```

Comment:
Assume this portion is going to be refactored. If not I couldn't follow the condition.